### PR TITLE
AWS Native Utilization Policy

### PR DESCRIPTION
**Pull request by rshade**
### Description

Creates a report of utilzation using native metrics

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Comment by rshade** [comment]
https://governance.rightscale.com/org/6/projects/7954/policy-incidents/5cc74eb7dd20d600713bbbf9

File: cost/aws/native_utilization/native_utilization_ap_south_1_no_cloud_in_rs [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+permission "perm_read_creds" do
+  actions   "rs_cm.show_sensitive","rs_cm.index_sensitive"
+  resources "rs_cm.credentials"
+end
+
+###############################################################################
+# Parameters
+###############################################################################
+
+parameter "param_email" do
```
Comment (cdwilhelm) [inline cost/aws/native_utilization/native_utilization_ap_south_1_no_cloud_in_rs]: Last week we added a new pattern for parameters.  All parameters should have a label and description.  The email address param needs to be changed and you probably need to add the exclude tag

```
parameter "param_email" do

  type "list"

  label "Email addresses to notify"

  description "Email addresses of the recipients you wish to notify when new incidents are created"

end


 


parameter "param_exclude_tags" do

  type "list"

  label "Tags to ignore"

  description "List of tags that will exclude resources from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag keys or key/value pairs can be listed. Example: 'test,env=dev'"

end
```

**Comment by rshade** [comment]
https://governance.rightscale.com/org/6/projects/7954/policy-incidents/5cc898821a0f81007190442b

File: cost/aws/native_utilization/aws_native_utilization.pt [hunk 2]
```diff
@@ -0,0 +1,676 @@
+name "AWS Native Utilization"
```
Comment (srpomeroy) [inline cost/aws/native_utilization/aws_native_utilization.pt]: Not sure if `Native` is the best way to explain this Policy Template.
AWS CloudWatch Instance Utilization?
AWS Instance Utilization?
...?

File: cost/aws/native_utilization/aws_native_utilization.pt [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+rs_pt_ver 20180301
+type "policy"
+short_description "Checks AWS Regions configured in RightScale and gets cloudwatch data for instances. See the [README](https://github.com/rightscale/policy_templates/tree/master/cost/native_utilization/) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
```
Comment (srpomeroy) [inline cost/aws/native_utilization/aws_native_utilization.pt]: `cloudwatch` should be `CloudWatch`

File: cost/aws/native_utilization/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+## AWS Native Utilization Policy
```
Comment (srpomeroy) [inline cost/aws/native_utilization/README.md]: Not sure if Native is the best way to explain this Policy Template.
AWS CloudWatch Instance Utilization?
AWS Instance Utilization?
...?

File: cost/aws/native_utilization/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+### What it does
+
+This Policy Template checks aws regions configured in RightScale and gets cloudwatch data for instances
```
Comment (srpomeroy) [inline cost/aws/native_utilization/README.md]: cloudwatch should be CloudWatch

File: cost/aws/native_utilization/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+### Prerequesites
+- The `credential_viewer`,`observer`,`actor` roles
```
Comment (srpomeroy) [inline cost/aws/native_utilization/README.md]: We have sections in other READMEs called `Required Permissions` or `Cloud Management Required Permissions`/`AWS Required Permissions`. We should mimic that layout here.
Example: https://github.com/rightscale/policy_templates/blob/master/operational/aws/vpc_name_sync/README.md

Comment (srpomeroy) [reply cost/aws/native_utilization/README.md]: What is the `actor` role required for?

File: cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt [hunk 7]
```diff
@@ -0,0 +1,666 @@
...
+end
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_utilization" do
+  validate_each $ds_merged_metrics do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing instance data"
```
Comment (srpomeroy) [inline cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt]: The summary should include more relevant info about the specific policy.
`summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing AWS instance CloudWatch utilization data"` ??

File: cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt [hunk 8]
```diff
@@ -0,0 +1,666 @@
+name "AWS Instance CloudWatch Utilization"
+rs_pt_ver 20180301
+type "policy"
+short_description "Checks AWS Regions configured in RightScale and gets CloudWatch data for instances. See the [README](https://github.com/rightscale/policy_templates/tree/master/cost/native_utilization/) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
```
Comment (srpomeroy) [inline cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt]: Should this be "AWS Instances"? We should also include a note about the timeframe calculated.
Example: "Reports on AWS instances CloudWatch Utilization over the past 30 days..."

File: cost/aws/instance_cloudwatch_utilization/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+## AWS Instance CloudWatch Utilization Policy
+
+### What it does
+
+This Policy Template checks aws regions configured in RightScale and gets CloudWatch data for instances
```
Comment (srpomeroy) [inline cost/aws/instance_cloudwatch_utilization/README.md]: Should include a note about the timeframe used to collect metrics and our recommendation on how often to run the policy based on that timeframe.

File: cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt [hunk 10]
```diff
@@ -0,0 +1,666 @@
...
+      {"region": "eu-west-3"},
+      {"region": "sa-east-1"},
+      {"region": "eu-north-1"}
+    )
+  EOS
+end
+
+script "js_instances", type: "javascript" do
+  result "results"
+  parameters "ds_instance_set"
```
Comment (avinashbhashyam-rs) [inline cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt]: isn't this region set?

File: cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt [hunk 11]
```diff
@@ -0,0 +1,666 @@
...
+      'Statistics.member.1': 'Maximum',
+      'Statistics.member.2': 'Average',
+      'Statistics.member.3': 'Minimum'
+    }
+  }
+EOS
+end
+
+script "js_merged_metrics", type: "javascript" do
+  parameters "ds_cloudwatch_cpu_maximum", "ds_cloudwatch_disk_usage", "ds_cloudwatch_mem_usage"
```
Comment (avinashbhashyam-rs) [inline cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt]: isn't this `ds_cloudwatch_cpu_maximum` just cpu usage?

Comment (rshade) [reply cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt]: Yes sir, I updated it to cpu_usage. I probably was trying only maximum at first.

**Pull request merged**